Function for Analyzing CSV Files of Rosetta Outputs

Salahub opened the pull request:
Here is a function which can read from csv files which have populated cells in the top left of the spreadsheet, and have titled columns. As well, this was written so that it can only make one set of comparisons at once. In other words you can compare as many columns of data from different methods as you want, but don't try to insert empty columns followed by a different set of data. It outputs another titled CSV where the columns correspond to the tau values, chi squared values, and chi squared p-values respectively.

I don't know why this isn't working on my system. There seems to be some issue with importing SciPy's stats package, and my computer doesn't recognize either of the functions which are supposed to be part of said package. If anyone knows why this is the case I'd love to hear an explanation!

Salahub committed:
PyRosetta Output Analysis Function

Salahub committed:
PyRosetta Analysis Function

Salahub committed:
At this point expect like three more PyRosetta analysis commits because I keep forgetting things

Salahub committed:
At this point this is just ridiculous. So many PyRosetta Analysis commits

mattsmart commented on models/tridimensional/data_kleinstiver/Analysis of PyRosetta Output.py:
```diff
@@ -0,0 +1,72 @@
+
+import scipy, scipy.stats.kendalltau, scipy.stats.chisquare
+
+def los_clnr(los):
```
how about csv_row_parser or csv_transform or csv_parser, this fn name makes me think of LA lol

data_parser might be good

mattsmart commented on models/tridimensional/data_kleinstiver/Analysis of PyRosetta Output.py:
```diff
@@ -0,0 +1,72 @@
+    as_ints = [comma_split[0]] + map(lambda (el): map(float, el), comma_split[1:])
+    return as_ints
+
+def sort_data(lst):
```
`data_sort(data_parsed)` is more clear (if you choose to go with data_parser above)

mattsmart commented on models/tridimensional/data_kleinstiver/Analysis of PyRosetta Output.py:
```diff
@@ -0,0 +1,72 @@
+        i_lst = []
+        for val_lst in just_vals:
+            i_lst.append(val_lst[i])
+        properly_sorted_lst.append(i_lst)
```
I think this replaces lines 24-29 but should be tested:
`data_transposed = [[just_vals[row][col] for row in xrange(len(just_vals[0]))] for col in xrange(len(titles))]`

If it does its also faster (append is really slow in python). xrange also faster than range when actual lists aren't needed.

mattsmart commented on models/tridimensional/data_kleinstiver/Analysis of PyRosetta Output.py:
```diff
@@ -0,0 +1,72 @@
+        properly_sorted_lst.append(i_lst)
+    sorted_dict = {}
+    for i in range(len(titles)):
+        sorted_dict[titles[i]] = properly_sorted_lst[i]
```
`sorted_dict = {titles[i]: data_transposed[i] for i in xrange(len(titles))}`

mattsmart commented on models/tridimensional/data_kleinstiver/Analysis of PyRosetta Output.py:
```diff
@@ -0,0 +1,72 @@
+        chitau_dict[k] = k_dict
+    return chitau_dict
+
+def main(dataname, ref_name, filename):
```
2 lines between fns

mattsmart commented on models/tridimensional/data_kleinstiver/Analysis of PyRosetta Output.py:
```diff
@@ -0,0 +1,72 @@
+
+import scipy, scipy.stats.kendalltau, scipy.stats.chisquare
+
```
2 lines between imports and fns

mattsmart commented on models/tridimensional/data_kleinstiver/Analysis of PyRosetta Output.py:
```diff
@@ -0,0 +1,72 @@
+
```
should be no newlines at top

mattsmart commented:
Minor comments and some speed tips. Will comment more later

mattsmart commented:
also can we change the script name to something cleaner? `results_analysis.py` works with `results_csv.py`

jrhawley commented on models/tridimensional/data_kleinstiver/Analysis of PyRosetta Output.py:
```diff
@@ -0,0 +1,72 @@
+    return sorted_dict
+        
+
+def get_tau_n_chi(sorted_data, exp_key):
```
I don't know if I like this function name :p

Salahub committed:
Suggested Changes to the Function for Analysis of PyRosetta CSV output

Salahub committed:
Real complete changes this time

Salahub commented:
I cleaned it up. Added most suggestions and changed some names. To clarify one thing, the "csv_parser" function is simply a helper for "sort_data," which is why I left the two with the same input name "los." Also For some reason my computer still doesn't want to import the stats package for scipy, but I tested every function I could. Any idea why that issue might be occurring?

Salahub committed:
Merge branch 'master' of https://github.com/igem-waterloo/uwaterloo-igem-2015

mattsmart commented:
@Salahub I don't think its a default package -- have you downloaded and installed it? if so, do you have more than one version of python?

mattsmart commented:
@Salahub Has this been fully tested yet?

Salahub commented:
@mattsmart No... I've been trying to sort out Python on my computer. I downloaded Python XY at the beginning of the term because everything seemed to indicate that included SciPy, NumPy, and a number of other useful packages. Before that I just had Python but I was sure to uninstall and remove everything before installing Python XY. Should I just uninstall and reinstall?

mattsmart commented:
@Salahub message me on hipchat when you're free

mattsmart commented on models/tridimensional/data_kleinstiver/Analysis of PyRosetta Output.py:
```diff
@@ -0,0 +1,67 @@
+import scipy, scipy.stats.kendalltau, scipy.stats.chisquare
```
from scipy.stats import kendalltau
from scipy.stats import chisquare

Salahub committed:
Analysis of PyRosetta again

Salahub commented:
In the latest commit I had to make some changes. This is now the tested function. However it must consume normalized data, i.e. data transformed to be on [0,1]. I realized my implementation of chi square was incorrect and so I changed it slightly and that took some work.

Salahub committed:
Hopefully the last

Salahub commented:
Hopefully that's the last commit. Just realized that some of the changes I thought I made weren't going through for some reason.

jrhawley commented:
Syntactically it looks fine to me. @mattsmart @alexanian if one of you wants to test it we can merge it.

This also has two versions of the file, one in the root, one in models/tridimensional/data_kleinstiver/ (both identical). We can either just merge it as is and quickly get rid of the root file, or if you want to get rid of it now @Salahub that would work

mattsmart commented:
I haven't tested, but to prerequisites to merging this:
- remove the root file as James said plz
- more sanitary filename -- e.g. results_analysis.py

Salahub committed:
Fixing some things

Salahub commented:
Okay, went ahead and did that. Sorry for the root file business, that was a miscopy on my end.

mattsmart commented:
My bad, there's one more thing: let's move the file out of data kleinstiver and into dock validation folder

Salahub committed:
Location changes

Salahub commented:
Thy bidding has been done. It is in the right folder now!

mattsmart commented:
LGTM, I'm gonna wait for @alexanian to comment though because she has a better idea of what use we need from this

alexanian commented:
gonna merge, working on this later today
